File: 0001-0099/0001.Two_Sum/1.Two_Sum.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */

const twoSum = (numbers, target) => {
  const hash = new Map();
  for (let i = 0; i < numbers.length; i++) {
    const remaining= target - numbers[i];
    if (hash.has(remaining)) return [i, hash.get(remaining)];
    hash.set(numbers[i], i);
  }
};

File: 0001-0099/0001.Two_Sum/1.Two_Sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i, value in enumerate(nums):
            remaining = target - value
            if remaining in seen:
                return [i, seen[remaining]]
            else:
                seen[value] = i
